application/bettle/bettle-jpa/src/main/java/com/bettle/track/jpa/repos/BaseRepository.java

```diff
@@ -0,0 +1,62 @@
+package com.bettle.track.jpa.repos;
+
+import com.bettle.track.entities.AbstractParentEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.NoRepositoryBean;
+import org.springframework.data.repository.query.Param;
+
+@NoRepositoryBean
+public interface BaseRepository<T extends AbstractParentEntity> extends CrudRepository<T, Long>, JpaRepository<T, Long> {
+
+    @Override
+    @Query("UPDATE #{#entityName} e SET e.active_flag = 0 WHERE e.id = :id")
```

Comment:
Can we move queries to resources folder and make use of annotation with fetches the query from classpath.